### DISCUSSION TITLE
ChitChatter running on my github site.

### DISCUSSION THREAD

**Discussion by ralphshinevar** [jeremyckahn/chitchatter#167] [General]
Okay, I have forked ChitChatter to my Github.  I changed the package.json file to my URL.  I am at:

https://github.com/ralphshinevar/chitchatter/settings/secrets/actions/new

How to do the secret?

**Comment by jeremyckahn** [comment]
The instructions [here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-set-ssh-private-key-deploy_key) and [here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-create-ssh-deploy-key) demonstrate how this can be done.

**Comment by ralphshinevar** [comment]
How do I run this on github?

ssh-keygen -t rsa -b 4096 -C "$(git config user.email)" -f gh-pages -N ""

**Comment by jeremyckahn** [comment]
You'll need to run that in your local command line environment, not on GitHub.

**Comment by ralphshinevar** [comment]
Okay I have the private key in https://github.com/ralphshinevar/chitchatter/settings/secrets/actions  Where to put the public key?

**Reply by jeremyckahn** [reply]
That's covered [here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-create-ssh-deploy-key):

> - Go to **Deploy Keys** and add your public key with the **Allow write access**
> - Go to **Secrets** and add your private key as `ACTIONS_DEPLOY_KEY`

**Comment by ralphshinevar** [comment]
Okay I got the public key in Deploy Keys

**Comment by ralphshinevar** [comment]
How do I get to the ChitChatter forked app?  https://github.com/ralphshinevar/chitchatter takes me to the files again.

**Reply by jeremyckahn** [reply]
You'll need to enable GitHub Actions at https://github.com/ralphshinevar/chitchatter/actions and then trigger a deployment at https://github.com/ralphshinevar/chitchatter/actions/workflows/deploy.yml like so: 

<img width="1477" alt="Screenshot 2023-09-27 at 2 50 08 PM" src="https://github.com/jeremyckahn/chitchatter/assets/366330/a0978d44-b56f-4fe4-bcdd-35b835382556">

Your instance should then be available at https://ralphshinevar.github.io/chitchatter.

**Reply by ralphshinevar** [reply]
I am getting:  There are no workflow runs yet.

**Comment by ralphshinevar** [comment]
It is running.

**Comment by ralphshinevar** [comment]
It says it failed.

**Reply by ralphshinevar** [reply]
I do not have a main branch, only develop.

**Comment by ralphshinevar** [comment]
Tried again failed again.

**Comment by ralphshinevar** [comment]
3rd time no good.  Do I need to edit deploy.yml?

**Reply by jeremyckahn** [reply]
I think you'll need to create a `main` branch off of your `develop` branch in order for the job to succeed.

**Reply by ralphshinevar** [reply]
Created a main branch; noticed my key name was different in deploy.yml.  Made the change.  Took a couple of tries to get it to use the updated deploy.yml file.  8th try was a charm!!!  I am running.

**Reply by jeremyckahn** [reply]
Yay! I'm glad to hear it @ralphshinevar. Thanks for letting me know!

**Comment by ralphshinevar** [comment]
Thanks for your help.

**Comment by ralphshinevar** [comment]
Hey Jeremy.  A new issue has emerged with my github chitchatter.  When I use your github chitchatter, open a public room and copy the link to give to someone else, it works.  The link works.  I do the same on my github chitchatter and the link gets a 404 error.  

![image](https://github.com/jeremyckahn/chitchatter/assets/56281781/3ab44e66-4a97-442c-ae9e-a9dc7ad94a5f)

**Reply by ralphshinevar** [reply]
Private rooms get the same error.

**Reply by jeremyckahn** [reply]
Hi @ralphshinevar, this looks like a bug in Chitchatter. I've filed [an issue](https://github.com/jeremyckahn/chitchatter/issues/168) and will prioritize a fix. Thanks for reporting this!

**Reply by ralphshinevar** [reply]
Hi Jeremy.  Keep in mind your github fork works, mine does not.

**Reply by jeremyckahn** [reply]
This is because https://chitchatter.im/ serves the app from the domain root rather than a sub-route like a GitHub Pages fork. This issue only affects Chitchatter instances served from a sub-route.

**Reply by jeremyckahn** [reply]
@ralphshinevar https://github.com/jeremyckahn/chitchatter/issues/168 has been fixed. Please update your fork to use the latest code from [my `main` branch](https://github.com/jeremyckahn/chitchatter/tree/main) and your links should start working.

**Comment by ralphshinevar** [comment]
@jeremyckahn I have pulled from your main branch.  I am not a github guru, is that all I need to do?

**Comment by ralphshinevar** [comment]
Tested it after the pull; got the 404 error.

**Comment by ralphshinevar** [comment]
Doing the deployment.  I think that is next?

**Comment by ralphshinevar** [comment]
Okay!!!  After the deployment it is working (routing).  Now to test it on some my testers (friends).

**Reply by jeremyckahn** [reply]
Great! Thanks for letting me know that it's working for you.

**Comment by ralphshinevar** [comment]
Thanks for your help Jeremy.  And for teaching more about GitHub.

**Comment by ralphshinevar** [comment]
Hold on  Jeremy.  I get in from the link on my PC.  But my phone, laptop and my buddy testing get the 404 error.  the url changes and there is a question mark in it.

**Comment by ralphshinevar** [comment]
https://ralphshinevar.github.io/?/chitchatter/public/4c5ee15a-3c19-46fe-b3ce-7490bdec7f7e

**Comment by ralphshinevar** [comment]
![image](https://github.com/jeremyckahn/chitchatter/assets/56281781/ee9db080-9a6b-421e-b0ba-3da1de7677a5)

**Comment by ralphshinevar** [comment]
Jeremy,

Is Chitchatter a Node JS app?  There is a place in my shared hosting
(Direct Admin) where I can install the app as a Node JS system.

Ralph

**Reply by jeremyckahn** [reply]
@ralphshinevar Chitchatter isn't a NodeJS app inasmuch as it does not require NodeJS to run. It does use NodeJS as a build tool, though. Once the app is built (either locally or via GitHub Actions), it can be served from any static asset server (in a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts)).
